WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/AddProductToOrderViewModel.swift

```diff
@@ -208,6 +254,30 @@ private extension AddProductToOrderViewModel {
             }
             .store(in: &subscriptions)
     }
+
+    /// Updates the product results predicate & triggers a new sync when search term changes
+    ///
+    func configureProductSearch() {
+        $searchTerm
+            .dropFirst() // Drop initial value
+            .debounce(for: .milliseconds(500), scheduler: DispatchQueue.main)
+            .removeDuplicates()
+            .sink { [weak self] newSearchTerm in
+                guard let self = self else { return }
+
+                if newSearchTerm.isNotEmpty {
+                    // When the search query changes, also includes the original results predicate in addition to the search keyword.
+                    let searchResultsPredicate = NSPredicate(format: "ANY searchResults.keyword = %@", newSearchTerm)
+                    let subpredicates = [self.resultsPredicate].compactMap { $0 } + [searchResultsPredicate]
+                    self.productsResultsController.predicate = NSCompoundPredicate(andPredicateWithSubpredicates: subpredicates)
+                } else {
+                    // Resets the results to the full product list when there is no search query.
+                    self.productsResultsController.predicate = self.resultsPredicate
+                }
+
+                self.syncingCoordinator.resynchronize(reason: newSearchTerm)
```

Comment:
is the reason a good way to pass the search term? Seems like a misuse but I'm not very familiar with the sync coordinator API 🤔

Reply:
Yeah, it's a way to pass the search term and determine how to do the sync (either a full product sync or a product search). I think this is a valid way to use the `reason` parameter, according to the `SyncingCoordinatorDelegate` protocol:

https://github.com/woocommerce/woocommerce-ios/blob/1887d1e13ec703b805a56608b456b44a0cae884f/WooCommerce/Classes/Tools/SyncCoordinator.swift#L9-L16

It's maybe a misuse to rely on it for the actual search term (rather than just deciding how to perform the sync). However, I tried relying on the view model's `searchTerm` variable for the actual search term but that variable isn't actually updated until after `resychronize()` is called here, so it ends up performing the wrong action (based on the previous value of `searchTerm` rather than the new value). I couldn't think of a better way to handle that here. 🤔

Reply:
Actually, I might have been mistaken about this part:

> I tried relying on the view model's searchTerm variable for the actual search term but that variable isn't actually updated until after resychronize() is called here

I thought that was happening but in further testing it doesn't seem to be an issue; it's possible this behavior was only happening before I added `removeDuplicates()` to `configureProductSearch()`. I'll double check that and push another commit to avoid using the `reason` parameter like this.

Reply:
Updated that in [0f059bb](https://github.com/woocommerce/woocommerce-ios/pull/6112/commits/0f059bb55f52caec21be6df1d57971944bd03643).